views/users.py

```diff
@@ -46,14 +52,44 @@ def user_avatar(user_id):
     return send_file('static/images/user.jpg')
 
 
-@blueprint.route('/forgot', methods=['GET', 'POST'])
+@blueprint.route('/password/forgot', methods=['GET', 'POST'])
 def user_password_forgot():
-    forgot_form = ForgotForm()
+    forgot_form = PasswordForgotForm()
     if forgot_form.validate_on_submit():
+        user = User.query.filter(func.lower(User.email) == func.lower(forgot_form.email.data)).first()
+        if user is not None:
```

Comment:
if user is None:
    render_template(blah, invalid_user=True)
    break
rest of the stuff

and the corresponding template stuff